### ISSUE TITLE
Keybinding/keyboard issues under GNOME

### ISSUE THREAD

**Issue by pbogdan** [NixOS/nixpkgs#75110]
## Issue description

I'm having a few issues with certain keybindings either not working or behaving strangely.

### Desktop switching

I have configured 9 virtual desktops with a corresponding `Super-N` binding to switch to the N-th workspace:

```
switch-to-workspace-1=['<Super>1']
switch-to-workspace-2=['<Super>2']
switch-to-workspace-3=['<Super>3']
switch-to-workspace-4=['<Super>4']
switch-to-workspace-5=['<Super>5']
switch-to-workspace-6=['<Super>6']
switch-to-workspace-7=['<Super>7']
switch-to-workspace-8=['<Super>8']
switch-to-workspace-9=['<Super>9']
```

This works for the most part with two issues:

- (some?) of the keybindings randomly stop working and instead of switching to the appropriate virtual desktop launch applications instead. For as much as it irritates me I haven't paid attention to which binding launches what as I immediately close those windows. So far seen a web browser, Nautilus, not sure if something else. Usually (perhaps always?) locking the screen and logging back in fixes the issue.
- the bindings only work for workspaces 1-6; I can't use `Super-7` - `Super-9` though I can switch to those with previous-next desktop bindings

### Can't input M-: in Emacs

Per the heading - I'm unable to produce `M-:` in Emacs running under GNOME. Depending on how exactly I try to input it it results in Emacs seeing different results:

- pressing keys in this order
  - Alt -> Shift -> ; results in M-;
  - Shift -> Alt -> ; results in :

with one modifier's input being lost.

I don't really know if this a GNOME issue but I'm a bit lost of how to try and troubleshoot this any further so any help would be appreciated.

### Steps to reproduce

Unknown.

## Technical details

Please run `nix run nixpkgs.nix-info -c nix-info -m` and paste the result.

```
- system: `"x86_64-linux"`
 - host os: `Linux 5.3.13, NixOS, 19.09.1457.dae3575cee5 (Loris)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3`
 - channels(root): `"nixos-19.09.1505.a7aa5dbb194"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Xorg GNOME session with GDM display manager.

**Comment by pbogdan** [comment]
At least the Emacs issue is now solved. For some reason, unbeknownst to me, I had `Alt-Shift` enabled under Gnome Tweaks -> Keyboard & Mouse -> Additional Layout Options -> Switching to another layout. Seems strange as I don't even have or had multiple layouts to switch between. Not sure how but perhaps I somehow enabled that option accidentally...

**Comment by hedning** [comment]
Gnome binds `Super-N` to `switch-application-n` by default (where `n` refers to the nth application in the dash). As far as I know having two actions bound to the same key combo results in only one of the actions being bound. And it's random from start to start which actions get there «first». So I'm fairly certain this is an upstream issue.

**Comment by pbogdan** [comment]
Thank you hedning - do you know if that would that affect the bindings switching back and forth within a single session, only interrupted by locking of the screen?

And if you think that might be the case do you know where I should open an issue upstream? Sorry, I'm not very familiar with the Gnome ecosystem.

**Comment by hedning** [comment]
Hmm, I wouldn't have guessed it would happen when just locking when dealing with just native bindings, but not quite sure.

Extensions will do a disable/enable loop on lock/unlock, so if you're running dash-to-dock, which also binds `Super-N` that could be it. (Not sure why the `switch-application` actions doesn't show up in gnome-control-center, and therefor doesn't offer to replace, though the schemas are available so that's probably not us either).

 Gnome-shell's issue tracker https://gitlab.gnome.org/GNOME/gnome-shell/issues/

**Comment by pbogdan** [comment]
Thank you very much @hedning for mentioning `switch-to-application`. I didn't see those somehow but knowing how to find them now I cleared them out and so far everything seems good - can switch to desktops past 6th and so far haven't seen any of the erratic behaviour of a binding doing two different things depending on.. well, I'm not still sure what :-D.

Will go ahead and close this now. Thank you again for the help - I honestly wasn't sure whether the issue could be with NixOS, GNOME itself or my own setup so I appreciate all the details.

PS. thanks for letting me know about dash-to-dock - did not know about that extensions but it is very nice indeed, after disabling its keybindings that is :-D

**Comment by hedning** [comment]
Glad to be of help, I've run into the same issues when making a gnome-shell extension :) For completeness I found an existing issue for this here: https://gitlab.gnome.org/GNOME/gnome-shell/issues/1093. 3.32 added these default shortcuts so I'm guessing you started seeing this after upgrading from 3.30.

Now gnome-control-center will still happily bind `Super-N` without notifying the user about the conflict, so that's probably a bug with gnome-shell/gnome-control-center. I filed a bug for this upstream: https://gitlab.gnome.org/GNOME/gnome-shell/issues/2006

For other nice extensions there's [switcher](https://github.com/daniellandau/switcher), which lets you search for widows/applications, and if you like tiling I can plug my own extension [paperwm](https://github.com/paperwm/PaperWM/) :p.
